packages/neon/neon_files/lib/widgets/browser_view.dart

```diff
@@ -87,12 +87,13 @@ class _FilesBrowserViewState extends State<FilesBrowserView> {
                         final uploadingTaskTiles = buildUploadTasks(tasksSnapshot.requireData, sorted);
 
                         return NeonListView(
-                          scrollKey: 'files-${pathSnapshot.requireData.join('/')}',
+                          scrollKey: 'files-${pathSnapshot.requireData.path}',
                           itemCount: sorted.length,
                           itemBuilder: (final context, final index) {
                             final file = sorted[index];
-                            final matchingTask = tasksSnapshot.requireData
-                                .firstWhereOrNull((final task) => _pathMatchesFile(task.path, file.name));
+                            final matchingTask = tasksSnapshot.requireData.firstWhereOrNull(
+                              (final task) => widget.bloc.path.value.join([file.name]) == task.path,
```

Comment:
It might make more sense to compare the path and file name separtely.
```suggestion
                              (final task) => file.name == task.path.name && widget.bloc.path.value == task.path.parent,
```
No need to join anything and early aborting if the file name mismatch (the most common)